cli/cmd/profile.go

```diff
@@ -87,3 +123,144 @@ func renderProfileTemplate(config *templateConfig, w io.Writer) error {
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s.%s.svc.cluster.local", options.name, options.namespace),
+			Namespace: controlPlaneNamespace,
+		},
+		TypeMeta: meta_v1.TypeMeta{
+			APIVersion: "linkerd.io/v1alpha1",
+			Kind:       "ServiceProfile",
+		},
+	}
+
+	routes := make([]*sp.RouteSpec, 0)
+
+	paths := make([]string, 0)
+	for path := range swagger.Paths.Paths {
+		paths = append(paths, path)
+	}
+	sort.Strings(paths)
+
+	for _, path := range paths {
+		item := swagger.Paths.Paths[path]
+		pathRegex := pathToRegex(path)
+		if item.Delete != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodDelete, item.Delete.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Get != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodGet, item.Get.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Head != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodHead, item.Head.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Options != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodOptions, item.Options.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Patch != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPatch, item.Patch.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Post != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPost, item.Post.Responses)
+			routes = append(routes, spec)
+		}
+		if item.Put != nil {
+			spec := mkRouteSpec(path, pathRegex, http.MethodPut, item.Put.Responses)
+			routes = append(routes, spec)
+		}
+	}
+
+	profile.Spec.Routes = routes
+	output, err := yaml.Marshal(profile)
+	if err != nil {
+		return fmt.Errorf("Error writing Service Profile: %s", err)
+	}
+	w.Write(output)
+
+	return nil
+}
+
+func mkRouteSpec(path, pathRegex string, method string, responses *spec.Responses) *sp.RouteSpec {
+	return &sp.RouteSpec{
+		Name:            fmt.Sprintf("%s %s", method, path),
+		Condition:       toReqMatch(pathRegex, method),
+		ResponseClasses: toRspClasses(responses),
+	}
+}
+
+func pathToRegex(path string) string {
+	escaped := regexp.QuoteMeta(path)
+	replaced := pathParamRegex.ReplaceAllLiteralString(escaped, "[^/]*")
+	return fmt.Sprintf("^%s$", replaced)
+}
+
+func toReqMatch(path string, method string) *sp.RequestMatch {
+	return &sp.RequestMatch{
+		Path:   path,
+		Method: method,
+	}
+}
+
+func toRspClasses(responses *spec.Responses) []*sp.ResponseClass {
+	if responses == nil {
+		return nil
+	}
+	classes := make([]*sp.ResponseClass, 0)
+
+	statuses := make([]int, 0)
+	for status := range responses.StatusCodeResponses {
+		statuses = append(statuses, status)
+	}
+	sort.Ints(statuses)
+
+	for _, status := range statuses {
+		cond := &sp.ResponseMatch{
+			Status: &sp.Range{
+				Min: uint32(status),
+				Max: uint32(status),
+			},
+		}
+		failure := false
+		if status >= 500 {
+			failure = true
+		}
+		classes = append(classes, &sp.ResponseClass{
+			Condition: cond,
+			IsFailure: !failure,
```

Comment:
Isn't this the inverse of what we want? If I use this example swagger definition:

```
swagger: "2.0"
info:
  title: Sample API
  description: API description in Markdown.
  version: 1.0.0
host: api.example.com
basePath: /v1
schemes:
  - https
paths:
  /users:
    get:
      summary: Returns a list of users.
      description: Optional extended description in Markdown.
      produces:
        - application/json
      responses:
        200:
          description: OK
```

Then I end up with:

```
    response_classes:
    - condition:
        status:
          max: 200
          min: 200
      is_failure: true
```

But shouldn't that be `is_failure: false`?

Reply:
Also, I think this would be easier to reason about if you condensed it a bit:

```diff
diff --git a/cli/cmd/profile.go b/cli/cmd/profile.go
index 391966ef..6ccce642 100644
--- a/cli/cmd/profile.go
+++ b/cli/cmd/profile.go
@@ -253,13 +253,9 @@ func toRspClasses(responses *spec.Responses) []*sp.ResponseClass {
 				Max: uint32(status),
 			},
 		}
-		failure := false
-		if status >= 500 {
-			failure = true
-		}
 		classes = append(classes, &sp.ResponseClass{
 			Condition: cond,
-			IsFailure: !failure,
+			IsFailure: status >= 500,
 		})
 	}
 	return classes
```